### DISCUSSION TITLE
Embla carousel not working properly until resizing window

### DISCUSSION THREAD

**Discussion by SocDario** [davidjerleke/embla-carousel#340] [Help] [answered]
Hello!
First of all I want to say that you did great job with Embla and I really like it.
However I encountered this problem in my Reactjs project, I tried to use horizontal drag free scrolling carousel. I have tried to reproduce this problem on codesandbox but somehow it's working properly there (same parts of code from project), but project is kinda too complicated to be fully reproduced on codesandbox and I saw few people here have had same issue so maybe we can resolve it this way. So the problem starts when I navigate from landingPage to page where slider is, slider is moveable but it comes back to it's initial position, until I inspected page to check what is going on and noticed that when window is resized slider started working properly.
So i was searching for my answer here on issues and discussions and found one guy that have had same issue as me and you solved it like this `<div css={styles.viewport} ref={isLoading?null: viewportRef}>` , I have tried that and not working for me, I also tried to combine it with useEffect and still not working. Also found about hard reseting Embla with this:

```
const [viewportRef, embla] = useEmblaCarousel({
    dragFree: true,
    containScroll: 'trimSnaps',
  });

  useEffect(() => {
    embla?.reInit();
  }, []);
```
That also didn't work for me. I know that it's complicated to solve problems like this when codesandbox isn't provided but I really can't reproduce that issue there, don't know why.. Any suggestions or options would help, thanks a lot!

EDIT:
I figured out that `containScroll: 'trimSnaps'` is causing my issue, but without that option I have unwanted whitespace on the left side of slider

**Comment by davidjerleke** [comment, marked as answer]
Hi @Freez1995,

This has been brought up before. Please read [this response](https://github.com/davidjerleke/embla-carousel/issues/195#issuecomment-907632926). I hope it helps.

Best,
David

**Reply by SocDario** [reply]
Problem solved with useEffect, setting "isLoaded" state to true after props are changed. Thanks for fast answer, Embla rocks!
